File: backend/schemas/review.py
from pydantic import BaseModel


class ReviewQuestion(BaseModel):

    id: str
    
    index: int

    question: str

    options: dict[str, str]

    userAnswer: str | None = None

    correctAnswer: str

    correctOption: str

    marked: bool

    explanation: str

    isCorrect: bool

    isSkipped: bool

class ReviewSummary(BaseModel):

    examId: str

    questionBank: str

    mode: str

    questionCount: int

    correct: int

    wrong: int

    unanswered: int

    percentage: float

    completedAt: str

    startedAt: str

    timed: bool

    durationMinutes: int | None = None


class ReviewResponse(BaseModel):

    examId: str

    summary: ReviewSummary

    questions: list[ReviewQuestion]

File: frontend/src/types/review.ts
export interface ReviewQuestion {
  id: string;

  index: number;
  
  question: string;

  options: Record<string, string>;

  userAnswer: string | null;

  correctAnswer: string;

  correctOption: string;

  marked: boolean;

  explanation: string;

  isCorrect: boolean;

  isSkipped: boolean;
}

export interface ReviewSummary {
  examId: string;

  questionBank: string;

  mode: string;

  questionCount: number;

  correct: number;

  wrong: number;

  unanswered: number;

  percentage: number;

  completedAt: string;

  startedAt: string;

  timed: boolean;

  durationMinutes: number | null;
}

export interface ReviewResponse {
  examId: string;

  summary: ReviewSummary;

  questions: ReviewQuestion[];
}